pkg/alibabacloud/eni/types/types.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"github.com/cilium/cilium/pkg/ipam/types"
+)
+
+// AlibabaCloudSpec is the ENI specification of a node. This specification is considered
+// by the cilium-operator to act as an IPAM operator and makes ENI IPs available
+// via the IPAMSpec section.
+//
+// The ENI specification can either be provided explicitly by the user or the
+// cilium agent running on the node can be instructed to create the CiliumNode
+// custom resource along with an ENI specification when the node registers
+// itself to the Kubernetes cluster.
+type AlibabaCloudSpec struct {
```

Comment:
Nit: I think we can rename this to just `Spec` because we already know that this struct is inside the alibaba package